App/src/test/java/vut/fit/gja2023/app/LoadingFromCsvTests.java

```diff
@@ -0,0 +1,120 @@
+                
+            } catch (IOException ex) {
+                Logger.getLogger(LoadingFromCsvTests.class.getName()).log(Level.SEVERE, null, ex);
+            } catch (IllegalArgumentException ex) {
+                Logger.getLogger(LoadingFromCsvTests.class.getName()).log(Level.SEVERE, null, ex);
+            } catch (IndexOutOfBoundsException ex) {
+                Logger.getLogger(LoadingFromCsvTests.class.getName()).log(Level.SEVERE, null, ex);
+            } catch (CsvException ex) {
+                Logger.getLogger(LoadingFromCsvTests.class.getName()).log(Level.SEVERE, null, ex);
+            }
+	}
+	
+        @Test
+	private void readCsvReverseTest() {
+            CsvReaderConfig config = new CsvReaderConfig(new int[]{1,0}, true);
+            try {
+                List<String[]> res = CsvReaderService.readCsv(TEST_FILE_ONE_PATH, config);
+                checkResult(TEAMS_RESULT_REVERSE, res);
+                
+            } catch (IOException ex) {
+                Logger.getLogger(LoadingFromCsvTests.class.getName()).log(Level.SEVERE, null, ex);
+            } catch (IllegalArgumentException ex) {
+                Logger.getLogger(LoadingFromCsvTests.class.getName()).log(Level.SEVERE, null, ex);
+            } catch (IndexOutOfBoundsException ex) {
+                Logger.getLogger(LoadingFromCsvTests.class.getName()).log(Level.SEVERE, null, ex);
+            } catch (CsvException ex) {
+                Logger.getLogger(LoadingFromCsvTests.class.getName()).log(Level.SEVERE, null, ex);
+            }
+        }
+        
+        private void checkResult(List<String[]> expected, List<String[]> res) {
+            for (int i = 0; i < expected.size(); i++) {
+                for (int j = 0; j < expected.get(0).length; j++) {
+                    Assertions.assertEquals(expected.get(i)[j], res.get(i)[j]);
+                }
+            }
+        }
+        
+        @Test
+	private void readCsvIndexMaxOutOfRangeTest() {
+            Assertions.assertThrows(IndexOutOfBoundsException.class, () -> {
+                CsvReaderConfig config = new CsvReaderConfig(new int[]{5,0}, true);
+                List<String[]> res = CsvReaderService.readCsv(TEST_FILE_ONE_PATH, config);
+            });
+        }
+
+        @Test
+	private void readCsvIndexMinOutOfRangeTest() {
+            Assertions.assertThrows(IndexOutOfBoundsException.class, () -> {
+                CsvReaderConfig config = new CsvReaderConfig(new int[]{1,-1}, true);
+                List<String[]> res = CsvReaderService.readCsv(TEST_FILE_ONE_PATH, config);
+            });
+        }
+        
+        @Test
+	private void readCsvColumnOrderSizeMismatchTest() {
+            Assertions.assertThrows(IllegalArgumentException.class, () -> {
+                CsvReaderConfig config = new CsvReaderConfig(new int[]{5,0, 1}, true);
+                List<String[]> res = CsvReaderService.readCsv(TEST_FILE_ONE_PATH, config);
+            });
```

Comment:
zmazat scope, List<String[]> res je zbytocne, netreba priradovat ked len testujeme ci vyhodi exception alebo ne,  tiež do assertThrows davaj vzdy len to konkretne volanie čo chces testovať, takže ani vytvaranie instancii nie, celé takto: aj pri zvysnych testov potom pls
```suggestion
    void readCsvIndexMaxOutOfRangeTest() {
        CsvReaderConfig config = new CsvReaderConfig(new int[]{5, 0}, true);
        assertThrows(IndexOutOfBoundsException.class, () -> CsvReaderService.readCsv(TEST_FILE_ONE_PATH, config));
    }
```